src/postprocess/band_structure.jl

```diff
@@ -30,39 +32,17 @@ function high_symmetry_kpath(model; kline_density=40)
                 kpath=[["-½", "½"]], kbranches=[1:length(kcoords)])
     end
 
-    # - Brillouin.jl expects the input direct lattice to be in the conventional lattice
-    #   in the convention of the International Table of Crystallography Vol A (ITA).
-    # - spglib uses this convention for the returned conventional lattice,
-    #   so it can be directly used as input to Brillouin.jl
-    # - The output k-Points and reciprocal lattices will be in the CDML convention.
-    conv_latt = get_spglib_lattice(model; to_primitive=false)
-    sgnum     = spglib_spacegroup_number(model)  # Get ITA space-group number
-    direct_basis   = Bravais.DirectBasis(collect(eachcol(conv_latt)))
-    primitive_latt = Bravais.primitivize(direct_basis, Bravais.centering(sgnum, 3))
-
-    primitive_latt ≈ collect(eachcol(model.lattice)) || @warn(
-        "DFTK's model.lattice and Brillouin's primitive lattice do not agree. " *
-        "The kpath selected to plot the band structure might not be most appropriate."
-    )
-
-    kp     = Brillouin.irrfbz_path(sgnum, direct_basis)
+    # TODO: Wrap this into a function? (Repeated pattern in spglib.jl)
+    spg_positions, spg_numbers, _ = spglib_atoms(model.atoms)
+    cell = Spglib.Cell(model.lattice, spg_positions, spg_numbers)
+
+    kp     = Brillouin.irrfbz_path(cell)
     kinter = Brillouin.interpolate(kp, density=kline_density)
 
     # TODO Need to take care of time-reversal symmetry here!
     #      See https://github.com/JuliaMolSim/DFTK.jl/pull/496/files#r725203554
 
-    # Need to double the points whenever a new path starts
-    # (for temporary compatibility with pymatgen)
-    # TODO Remove this later
-    kcoords = empty(first(kinter.kpaths))
-    for kbranch in kinter.kpaths
-        idcs = findall(k -> any(sum(abs2, k - kcomp) < 1e-5
-                                for kcomp in values(kp.points)), kbranch)
-        @assert length(idcs) ≥ 2
-        idcs = idcs[2:end-1]  # Don't duplicate first and last
-        idcs = sort(append!(idcs, 1:length(kbranch)))
-        append!(kcoords, kbranch[idcs])
-    end
+    kcoords = vcat(kinter.kpaths...)
 
     T = eltype(kcoords[1])
     klabels = Dict{String,Vector{T}}(string(key) => val for (key, val) in kp.points)
```

Comment:
Similarly here, I don't see the reason to convert this to string. We needed that for pymatgen compatibility, but we might as well keep the symbol now.

Reply:
Done. Now the Symbol to String conversion is done in `kdistances_and_ticks`.